token.go

```diff
@@ -49,7 +49,30 @@ func (c *Connection) Tokens() (access, refresh string, err error) {
 // TokensContext returns the access and refresh tokens that is currently in use by the connection.
 // If it is necessary to request a new token because it wasn't requested yet, or because it is
 // expired, this method will do it and will return an error if it fails.
+// The function will retry the operation in an exponential-backoff method.
 func (c *Connection) TokensContext(ctx context.Context) (access, refresh string, err error) {
+	operation := func() error {
+		c.logger.Debug(ctx, "trying to get tokens")
+		var code int
+		code, access, refresh, err = c.tokensContext(ctx)
+		if err != nil {
+			if code >= http.StatusInternalServerError {
+				c.logger.Error(ctx, "failed to get tokens, got http code %d, will attempt to retry. err: %v", code, err)
+				return err
+			}
+			c.logger.Debug(ctx, "failed to get tokens, got http code %d, will not attempt to retry. err: %v", code, err)
+			return nil
```

Comment:
@tzvatot are you dont want to retry even if `err` is non-nil?

Reply:
@nimrodshn see @jhernand comment above:
```
We should retry this only in situations where we know that retry may help
```
Currently we suspect that only http 5xx can be recovered by a retry. Not sure about other errors so not retrying for those.

Reply:
Alright. :+1: